### ISSUE TITLE
Code usage accuracy

### ISSUE THREAD

**Issue by akelyasir** [photino/zino#44]
Hi @photino 

Sorry for asking so many questions. I am trying to learn your system and usage logic. I want to share a code that I use with you. Is it a correct method? Or I want to understand if you have a different use.

**/// Controller**

`
pub async fn new(mut req: Request) -> Result {
    let mut cluster = AWSEC::new();
    
    let mut res = req.model_validation(&mut cluster).await?;
    let validation = cluster.check_constraints().await.extract(&req)?;
    if !validation.is_success() {
        return Err(Rejection::bad_request(validation).context(&req).into());
    }

    // If cloud_provider is not `Generic`
    let cluster_id = cluster.id().to_string();
    let id = cluster.id().to_owned();
    
    cluster.insert().await.extract(&req)?;

    let mut body = Map::new();
    body.insert("cidr_block".into(), "CIDR BLOCK".into());

    let _asd = service::aws::execute_aws_vpc(id, body).await.extract(&req)?;

    let data = json!({
        "method": req.request_method().as_ref(),
        "path": req.request_path(),
        "cluster_id": cluster_id,
    });
            
    res.set_code(StatusCode::CREATED);
    res.set_data(&data);

    Ok(res.into())
}
`

**/// Services**

`pub async fn execute_aws_vpc(id: Uuid, body: Map) -> Result<(), Error> {

    let (_v, m) = AWSEC::update_by_id(&id, body).await?;
    drop(m);
    
    Ok(())
}`

While adding a new record, I tried to update the model added with the incoming data by performing an api operation on aws. There is no problem in the working of the code. Updating the new table by updating the body value in the controller. Is this approach the right one for your system? Or should a different method be followed?

Thanks and regards.

**Comment by photino** [comment]
Yes, it is the right approach to define customized controllers and services. It seems that you need to update the `cidr_block` field of the `AWSEC` model. Please note that you can also update the model by calling a setter before inserting it into the table:
```rust
cluster.set_cidr_block("CIDR BLOCK");
cluster.insert().await.extract(&req)?;
```
where the method `set_cidr_block` is defined in the `impl AWSEC` block.

**Comment by akelyasir** [comment]
Thanks @photino It is simpler and more convenient to define the function impl AWSEC {}. 

Thank you for your help and interest.
